plugins/inventory/nmap.py

```diff
@@ -135,8 +139,15 @@ def parse(self, inventory, loader, path, cache=True):
         if not user_cache_setting or cache_needs_update:
             # setup command
             cmd = [self._nmap]
+
+            if self._options['sudo']:
+                cmd.insert(0, 'sudo')
+
             if not self._options['ports']:
-                cmd.append('-sP')
+                cmd.append('-sV')
+                cmd.append('-p 22')
```

Comment:
I'm pretty sure that this will **not** work. I think you need
```suggestion
                cmd.extend(['-p', '22'])
```
In any case, this is a **breaking change** and won't get merged anyway. Add more options instead to control this behavior.

(Also, this should really **not** be part of this PR, but a separate PR. Mixing completely different things in one PR will mainly make sure that it takes a lot longer to get these things into a release.)

Reply:
Thanks for reviewing, and you are right I should do 2 PR one for sudo and one for ssh port scanning behaviour